### DISCUSSION TITLE
Get selected face in viewer

### DISCUSSION THREAD

**Discussion by stromhaug-tommy** [Macad3D/Macad3D#33] [General]
Dear Tobias,

When selecting a face in the wiever, how do i get the reference to the face selected? I am getting a wire.

**Comment by dipts** [comment]
Can you tell a little bit more about the context in which you find yourself?
Generally you use the function `GetSubshapeReference` on the shape. There is a script example in the [documentation](https://macad3d.net/userguide/go?guid=8bd40f7f-0687-46fe-ba84-2faf6bc91571) on the subject of subshape selection in the viewer and conversion to a reference.

**Reply by stromhaug-tommy** [reply]
My bad. I implemented my own viewer based on your help.

I present the object with this AisContext.Display(new AIS_Shape(shape), 1, 3, false);. for face selection.
Then i am using this in my selection event ;
TopoDS_Shape detectedShape = Macad.Occt.Helper.Ais.GetDetectedShapeFromContext(AisContext);
But here i get a wire and not a face.

**Reply by dipts** [reply]
`GetDetectedShapeFromContext` gives back the _detected_ shape, not the _selected_. Since Macad has it's own selection handlung it does not work with the selection functionality of AIS.
